lib/tests/config_test.cc

```diff
@@ -116,39 +116,45 @@ TEST_CASE("correct exceptions should be thrown", "[config]") {
 }
 
 TEST_CASE("should correctly parse durations", "[config]") {
-    SECTION("suffix strings should be correctly converted to nanoseconds") {
-        REQUIRE(2 == config::parse_duration("2", fake_origin(), "millis").count());
-        REQUIRE(2 == config::parse_duration("2ms", fake_origin(), "millis").count());
-        REQUIRE(2 == config::parse_duration("2millis", fake_origin(), "millis").count());
-        REQUIRE(2 == config::parse_duration("2millisecond", fake_origin(), "millis").count());
-        REQUIRE(2000 == config::parse_duration("2s", fake_origin(), "seconds").count());
-        REQUIRE(2000 == config::parse_duration("2seconds", fake_origin(), "seconds").count());
-        REQUIRE(120000 == config::parse_duration("2m", fake_origin(), "minutes").count());
-        REQUIRE(120000 == config::parse_duration("2minutes", fake_origin(), "minutes").count());
-        REQUIRE(7200000 == config::parse_duration("2h", fake_origin(), "hours").count());
-        REQUIRE(7200000 == config::parse_duration("2 hours", fake_origin(), "hours").count());
-        REQUIRE(172800000 == config::parse_duration("2d ", fake_origin(), "days").count());
-        REQUIRE(172800000 == config::parse_duration(" 2days", fake_origin(), "days").count());
+    SECTION("suffix strings should be correctly converted to appropriate durations") {
+        REQUIRE(2 == config::parse_duration("2", fake_origin(), "millis").total_milliseconds());
+        REQUIRE(2 == config::parse_duration("2ms", fake_origin(), "millis").total_milliseconds());
+        REQUIRE(2 == config::parse_duration("2millis", fake_origin(), "millis").total_milliseconds());
+        REQUIRE(2 == config::parse_duration("2millisecond", fake_origin(), "millis").total_milliseconds());
+        REQUIRE(2 == config::parse_duration("2s", fake_origin(), "seconds").seconds());
+        REQUIRE(2 == config::parse_duration("2seconds", fake_origin(), "seconds").seconds());
+        REQUIRE(2 == config::parse_duration("2m", fake_origin(), "minutes").minutes());
+        REQUIRE(2 == config::parse_duration("2minutes", fake_origin(), "minutes").minutes());
+        REQUIRE(2 == config::parse_duration("2h", fake_origin(), "hours").hours());
+        REQUIRE(2 == config::parse_duration("2 hours", fake_origin(), "hours").hours());
+        REQUIRE(48 == config::parse_duration("2d ", fake_origin(), "days").hours());
+        REQUIRE(48 == config::parse_duration(" 2days", fake_origin(), "days").hours());
```

Comment:
Can you add some tests around bounds checking? Like 2^63 nanoseconds and some large number of days?